crates/swc_ecma_minifier/src/compress/pure/strings.rs

```diff
@@ -427,21 +448,23 @@ impl Pure<'_> {
                     );
 
                     if let Some(cooked) = &mut r_first.cooked {
-                        *cooked =
-                            format!("{}{}", convert_str_value_to_tpl_cooked(&ls.value), cooked)
-                                .into()
+                        let str_part = convert_str_value_to_tpl_cooked(&ls.value);
```

Comment:
This allocates, so it's better to call `str_part.reserve(cooked.len());` after this line and push the string to `str_part`